### PR TITLE
HPCC-17216 Output each statistic on a single line (in batch mode)

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by ghalliday** [comment]
@shamser please review.
This makes locating slow queries (or queries that use lots of memory) much easier

**Comment by ghalliday** [comment]
Tested by running on the regression suite and checking stats output file.

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -9031,7 +9031,7 @@ extern WORKUNIT_API void exportWorkUnitToXMLFile(const IConstWorkUnit *wu, const
             IPropertyTree * stats = p->queryPropTree("Statistics");
             if (stats)
             {
-                saveXML(statsFilename, stats, 0, XML_Format|XML_SortTags|extraXmlFlags);
+                saveXML(statsFilename, stats, 0, (XML_Format|XML_SortTags|extraXmlFlags) & ~XML_LineBreakAttributes);
```
Comment (shamser) [inline common/workunit/workunit.cpp]: @ghalliday Doesn't this unset the XML_LineBreakAttribute bit?

**Review by shamser — COMMENTED** [review]
@ghalliday The code looks like it is unsetting rather than setting the XML_LineBreakAttribute bit?

File: common/workunit/workunit.cpp (hunk 1 above)
Comment (richardkchapman) [inline common/workunit/workunit.cpp]: I think that is the intention (no linebreaks between attributes...)

Comment (ghalliday) [inline common/workunit/workunit.cpp]: @shamser Yes - that is what I want to do, have all the attributes on a single line

Comment (ghalliday) [inline common/workunit/workunit.cpp]: XML_Format is a combination of multiple bits.

**Review by shamser — APPROVED** [review]
@richardkchapman Code reviewed ok.

File: common/workunit/workunit.cpp (hunk 1 above)
Comment (shamser) [inline common/workunit/workunit.cpp]: Should get my eyes tested (misread the ticket)

**Pull request merged**